applications/portfolioDrafts/getPortfolioDrafts.ts

```diff
@@ -0,0 +1,70 @@
+import { ScanCommand, ScanCommandInput } from "@aws-sdk/lib-dynamodb";
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
+import { ErrorCodes } from "./models/Error";
+import { PortfolioDraftSummary } from "./models/PortfolioDraftSummary";
+import { dynamodbDocumentClient as client } from "./utils/dynamodb";
+import { ApiSuccessResponse, ErrorResponse, ErrorStatusCode, SuccessStatusCode } from "./utils/response";
+
+const TABLE_NAME = process.env.ATAT_TABLE_NAME;
+const QUERY_PARAM_INVALID = new ErrorResponse(
+  { code: ErrorCodes.INVALID_INPUT, message: "Invalid request parameter" },
+  ErrorStatusCode.BAD_REQUEST
+);
+
+/**
+ * Convert string to integer, return defaultInt if undefined or NaN.
+ * @param str - The string value to convert
+ * @param defaultInt - The default integer value to return if str undefined or NaN
+ */
+function getIntegerOrDefault(str: string | undefined, defaultInt: number): number {
+  if (str === undefined) return defaultInt;
+  const int = parseInt(str);
+  if (isNaN(int)) return defaultInt;
+  return int;
+}
+
+/**
+ * Gets all Portfolio Drafts, TODO: "...to which the user has read access"
+ * Revisit once authentication and authorization are in place.
+ * @param event - The GET request from API Gateway
+ */
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+  // Optional query param 'offset' must be integer with minimum value of 0.
+  // Offset is the number of items to skip before starting to collect the result set.
+  const offset: number = getIntegerOrDefault(event.queryStringParameters?.offset, 0);
```

Comment:
Another option which relies on ECMAScript 2015 default variables:

```
function getOffset(offsetParam = "0") : number {
  return parseInt(offsetParam);
}
```

```
  const params: ScanCommandInput = {
    TableName: TABLE_NAME,
    Limit: getOffset(event.queryStringParameters?. offset),
  };
```

And finally, the same as above but with a one-liner that uses an anonymous function: 

```
    const params: ScanCommandInput = {
        TableName: TABLE_NAME,
        Limit: function (offset: string = "0") {return parseInt(offset);}(event.queryStringParameters?.offset),
    };
```

Reply:
This is probably too cute since it doesn't actually validate the parameter appropriately. But worth noting the existence of default function parameters.

Reply:
ECMAScript 2015 should be fine since TypeScript should transpile anyway.